[Bug] 2.5.1 does not compile for iOS: 'value(forHTTPHeaderField:)' requires iOS 13, package declares .iOS(.v12)

## Summary

`2.5.1` fails to compile for any iOS consumer. `Poller.swift` uses `HTTPURLResponse.value(forHTTPHeaderField:)`, introduced in iOS 13.0 / macOS 10.15, while `Package.swift` still declares `.iOS(.v12)`.

```
Sources/UnleashProxyClientSwift/Poller/Poller.swift:206:43: error: 'value(forHTTPHeaderField:)' is only available in iOS 13.0 or newer
```

Introduced by #137 (case-insensitive etag lookup), released in 2.5.1. `main` has the same code, so 2.5.0 is currently the last version that builds for iOS.

## Reproduction

Against tag `2.5.1`:

```sh
git clone --depth 1 --branch 2.5.1 https://github.com/Unleash/unleash-ios-sdk
cd unleash-ios-sdk

swift build
# Build complete! (8.78s)

xcodebuild -scheme UnleashProxyClientSwift -destination 'generic/platform=iOS' build
# Poller.swift:206:43: error: 'value(forHTTPHeaderField:)' is only available in iOS 13.0 or newer
```

Also reproduces when the package is consumed from an Xcode project targeting an iOS simulator.

## Why CI did not catch it

All three jobs in `.github/workflows/build.yml` run `swift build` / `swift test` on `macos-latest`, which compiles for the host platform. Against the declared `.macOS(.v10_15)` minimum the API is exactly available, so the build passes. No job ever compiles against the `.iOS(.v12)` minimum, so the availability violation is invisible.

## Suggested fix

Either:

- raise the declared platform to `.iOS(.v13)` (and `.macOS(.v10_15)` already matches), or
- read the header via `allHeaderFields` with a case-insensitive lookup, keeping the iOS 12 floor.

Plus a CI job that builds for an iOS destination, so the declared minimum is actually exercised.

--- a/Sources/UnleashProxyClientSwift/Poller/Poller.swift
+++ b/Sources/UnleashProxyClientSwift/Poller/Poller.swift
@@ -203,7 +203,7 @@ public class Poller {
 
             var result: FeatureResponse?
 
-            if let newEtag = httpResponse.value(forHTTPHeaderField: "Etag"), !newEtag.isEmpty {
+            if let newEtag = value(forHTTPHeaderField: "Etag", in: httpResponse), !newEtag.isEmpty {
                 lock.lock()
                 self.etag = newEtag
                 lock.unlock()
@@ -247,6 +247,14 @@ public class Poller {
         return lowercasedHeader == "content-type" ||
             lowercasedHeader == "if-none-match" ||
             lowercasedHeader.hasPrefix("unleash-")
+    }
+
+    private func value(forHTTPHeaderField field: String, in response: HTTPURLResponse) -> String? {
+        // Once iOS 12 support is dropped, replace this with HTTPURLResponse.value(forHTTPHeaderField:).
+        return response.allHeaderFields.first { key, _ in
+            guard let header = key as? String else { return false }
+            return header.caseInsensitiveCompare(field) == .orderedSame
+        }?.value as? String
     }
 }
 

--- a/Tests/UnleashProxyClientSwiftTests/FakeHTTPResponse.swift
+++ b/Tests/UnleashProxyClientSwiftTests/FakeHTTPResponse.swift
@@ -1,6 +1,6 @@
 import Foundation
 
-/// Simulates a real network `HTTPURLResponse` by keeping header keys case-insensitive
+/// Simulates a real network `HTTPURLResponse` with raw header casing
 /// without Apple's default constructor key-canonicalization.
 final class FakeHTTPURLResponse: HTTPURLResponse, @unchecked Sendable {
     private let rawHeaders: [String: String]
@@ -16,11 +16,5 @@ final class FakeHTTPURLResponse: HTTPURLResponse, @unchecked Sendable {
 
     override var allHeaderFields: [AnyHashable: Any] {
         return rawHeaders
-    }
-
-    override func value(forHTTPHeaderField field: String) -> String? {
-        return rawHeaders.first { key, _ in
-            key.caseInsensitiveCompare(field) == .orderedSame
-        }?.value
     }
 }

--- a/Tests/UnleashProxyClientSwiftTests/PollerTests.swift
+++ b/Tests/UnleashProxyClientSwiftTests/PollerTests.swift
@@ -107,6 +107,17 @@ final class PollerTests: XCTestCase {
         XCTAssertEqual(poller.etag, "W/\"710-wJiNH+MQpj0ruMo7n/9j36tB+Fg\"")
     }
 
+    func testRawEtagResponseHeaderLookupIsCaseInsensitive() {
+        let response = mockResponse(headerFields: ["ETAG": "W/\"710-wJiNH+MQpj0ruMo7n/9j36tB+Fg\""])
+        let data = stubData()
+        let session = MockPollerSession(data: data, response: response)
+        let poller = createPoller(with: session)
+
+        XCTAssertTrue(poller.etag.isEmpty)
+        poller.getFeatures(context: Context())
+        XCTAssertEqual(poller.etag, "W/\"710-wJiNH+MQpj0ruMo7n/9j36tB+Fg\"")
+    }
+
     func testEmptyEtagResponseHeader() {
         let response = mockResponse(headerFields: ["Etag": ""])
         let data = stubData()

